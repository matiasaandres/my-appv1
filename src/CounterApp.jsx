import { useState } from 'react';
import PropTypes from 'prop-types';

import React from 'react';


const CounterApp = ({value}) => {

    const [counter,setCounter] = useState(0);

    const  handleAdd = ()=>{
     
        setCounter(counter+1);

    }
    const handleSubtract = () =>{
        setCounter(counter-1)
    }
    const handleReset = ()=>{
        setCounter(value)
    }
  return (
    <>

    <h1>CounterApp</h1>
    <h2>{counter}</h2>
    <button className='btn  btn-primary' onClick={handleAdd}>+1</button>
    <button className='btn  btn-danger' onClick={handleSubtract}>-1</button>
    <button className='btn  btn-warning' onClick={handleReset}>Reset</button>

    </>
    
  )
}

export default CounterApp;

CounterApp.propTypes = {
    value: PropTypes.number.isRequired
}