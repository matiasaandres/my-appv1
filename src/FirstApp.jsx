import PropTypes from 'prop-types';

const FirstApp = ({title, subtitle, name}) =>{

    return(

        <>
        <h1>{title}</h1>
        
        <p>{subtitle}</p>
        <p>{name}</p>
        </>
    )
}
export default FirstApp;

FirstApp.propTypes = {
    title: PropTypes.string.isRequired,
    subtitle: PropTypes.string
}

FirstApp.defaultProps = {
    name: 'Matías Morales',
    subtitle: 'No hay subtítulo'
}