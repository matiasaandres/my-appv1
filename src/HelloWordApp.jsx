export function App(){
    return (<h1>Hola Mundo</h1>);
}
export const hola = '123';