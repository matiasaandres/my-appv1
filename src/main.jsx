import React from  'react';
import { ReactDOM } from 'react-dom/client';

import {App, hola} from './HelloWordApp';

ReactDOM.createRoot(document.getElementById('root')).render(
    <React.StrictMode>
        <App />
    </React.StrictMode>
)