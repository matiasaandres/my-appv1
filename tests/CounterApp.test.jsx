import { fireEvent, render, screen } from "@testing-library/react";
import { CounterApp } from "../src/CounterApp";

describe("Prueba en <CounterApp />", () => {
  const initialValue = 10;

  test("debe hacer mach con el snapshot", () => {
    const { container } = render(<CounterApp value={initialValue} />);
    expect(container).toMatchSnapshot();
  });

  test("Debe mostrar el valor inicial de 100 ", () => {
    render(<CounterApp value={100} />);
    //expect(screen.getByText('100')).toBeTruthy();
    expect(screen.getByRole("heading", { level: 2 }).innerHTML).toContain(
      "100"
    );
  });

  test("Debe incrementar el con el boton +1", () => {
    render(<CounterApp value={initialValue} />);
    //fireEvent permite lanzar un evento
    fireEvent.click(screen.getByText("+1"));
    expect(screen.getByText("11")).toBeTruthy();
  });

  test("Debe decrementar el con el boton -1", () => {
    render(<CounterApp value={initialValue} />);
    //fireEvent permite lanzar un evento
    fireEvent.click(screen.getByText("-1"));
    //screen.debug;
    expect(screen.getByText("9")).toBeTruthy();
  });

  test("Debe de funcionar el boton Reset", () => {
    render(<CounterApp value={initialValue} />);
    //fireEvent permite lanzar un evento
    fireEvent.click(screen.getByText("+1"));
    fireEvent.click(screen.getByText("+1"));
    fireEvent.click(screen.getByText("+1"));
    //fireEvent.click(screen.getByText("Reset"));
    //mejorar especificacion de boton
    fireEvent.click(screen.getByRole('button',{name:'btn-reset'}));
    expect(screen.getByText("10")).toBeTruthy();
  });
});
