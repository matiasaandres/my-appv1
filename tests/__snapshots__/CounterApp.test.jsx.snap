// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba en <CounterApp /> debe hacer mach con el snapshot 1`] = `
<div>
  <h1>
    CounterApp
  </h1>
  <h2>
     
    10
     
  </h2>
  <button>
    +1
  </button>
  <button>
     -1 
  </button>
  <button
    aria-label="btn-reset"
  >
     Reset 
  </button>
</div>
`;
